src/ImmersedBoundaries/immersed_reductions.jl

```diff
@@ -20,6 +20,11 @@ const IF = AbstractField{<:Any, <:Any, <:Any, <:ImmersedBoundaryGrid}
 @inline condition_operand(func::Function,         op::IF, cond,      mask) = ConditionalOperation(op; func, condition=NotImmersed(cond),     mask)
 @inline condition_operand(func::Function,         op::IF, ::Nothing, mask) = ConditionalOperation(op; func, condition=NotImmersed(truefunc), mask)
 @inline condition_operand(func::typeof(identity), op::IF, ::Nothing, mask) = ConditionalOperation(op; func, condition=NotImmersed(truefunc), mask)
+@inline function condition_operand(func::Function, op::IF, cond::AbstractArray, mask)
+    condition = NotImmersed(arch_array(architecture(op.grid), cond))
```

Comment:
```suggestion
    arch = architecture(op.grid)
    arch_condition = arch_array(arch, condition)
    ni_condition = NotImmersed(arch_condition)
```


Pointers:

* don't use full and abbreviated forms like `cond` vs `condition` to imply a difference between two objects. Be specific about the difference between these two.
* default towards doing just one thing in one line